pkg/anthropic/request.go

```diff
@@ -116,8 +118,8 @@ func NewToolResultContentBlock(toolUseID string, content interface{}, isError bo
 
 // ToolChoice specifies the tool preferences for a message request.
 type ToolChoice struct {
-	Type string `json:"type"` // Type of tool choice: "tool", "any", or "auto".
-	Name string `json:"name"` // Name of the tool to be used (if type is "tool").
+	Type string `json:"type"`           // Type of tool choice: "tool", "any", or "auto".
+	Name string `json:"name,omitempty"` // Name of the tool to be used (if type is "tool").
```

Comment:
omitempty required here so that explicitly passing "auto" or "any" for the tool choice type causing lead to an invalid request.